### PR TITLE
Language label for the language switcher dropdown

### PR DESCRIPTION
**Pull request by distantnative**
## This PR …

- [x] Support to edit in language dialog

### Enhancement
- Language: customize the `label` of the Panel's languages dropdown for the currently selected language

### CODE REVIEW

**Review by lukasbestle — COMMENTED** [review]
TBH I'd keep the "translate" icon. One could still hide it via CSS if it doesn't fit.

**Comment by afbora** [comment]
What do you think about also new optional `label` field for the language dialogs? Maybe it should be sync with code field as default.

image (4)

**Comment by distantnative** [comment]
Added

Screenshot 2023-10-19 at 18 54 58

File: tests/Cms/Languages/LanguageTest.php [hunk 1]
```diff
@@ -281,6 +281,25 @@ public function testIsDefault()
 		$this->assertFalse($language->isDefault());
 	}
 
+	/**
+	 * @covers ::label
+	 */
+	public function tesLabel()
```
Comment (lukasbestle) [inline tests/Cms/Languages/LanguageTest.php]:
```suggestion
	public function testLabel()
```

**Review by lukasbestle — COMMENTED** [review]
If the label is left blank, the fallback to the code does not work.

Two possible solutions I can think of:

- Change `Language::update()` to replace an empty string for the label with `null`. This way, a dev could still manually set an empty string in the language file. But if changed via the UI, leaving the field blank will make it fall back to the code.
- In the frontend, use a similar logic like with the code and locale fields (when changing the name or code fields, automatically update the label fields with the uppercase code). This one would probably be the most flexible as then the label could also be manually set to an empty string from the UI.

File: src/Cms/Language.php [hunk 2]
```diff
@@ -59,6 +60,7 @@ public function __construct(array $props)
 		$this->code         = trim($props['code']);
 		$this->default      = ($props['default'] ?? false) === true;
 		$this->direction    = ($props['direction'] ?? null) === 'rtl' ? 'rtl' : 'ltr';
+		$this->label        = trim($props['label'] ?? Str::upper($this->code));
```
Comment (afbora) [inline src/Cms/Language.php:63]: This line is not working correctly. The `label` prop from the frontend is always sent as empty string. So if user doesn't enter label, language dropdown shows empty string.

Should be empty check for `$props['label']` instead null check or `label` argument should be sent as `null` if not entered.

Comment (lukasbestle) [reply src/Cms/Language.php:63]: Yep, also ran into that. See the possible solutions in the comment above.

Comment (afbora) [reply src/Cms/Language.php:63]: Oh, I missed that.

**Comment by distantnative** [comment]
Closing this for now until it's ready

**Pull request closed without merging**